metron-platform/metron-writer/src/main/java/org/apache/metron/writer/BulkWriterComponent.java

```diff
@@ -115,15 +116,37 @@ public void commit(BulkWriterResponse response) {
   }
 
   public void error(String sensorType, Throwable e, Iterable<Tuple> tuples, MessageGetStrategy messageGetStrategy) {
+
+    if(!Iterables.isEmpty(tuples)) {
+      LOG.error("Failing tuples; count={}, error={}", Iterables.size(tuples), ExceptionUtils.getRootCauseMessage(e));
+    }
     tuples.forEach(t -> collector.ack(t));
     MetronError error = new MetronError()
             .withSensorType(sensorType)
             .withErrorType(Constants.ErrorType.INDEXING_ERROR)
             .withThrowable(e);
+    tuples.forEach(t -> error.addRawMessage(messageGetStrategy.get(t)));
+    ErrorUtils.handleError(collector, error);
+  }
+
+  /**
+   * Error a set of tuples that may not contain a valid message.
+   *
+   * <p>Without a valid message, the source type is unknown.
+   * <p>Without a valid message, the JSON message cannot be added to the error.
+   *
+   * @param e The exception that occurred.
+   * @param tuples The tuples to error that may not contain valid messages.
+   */
+  public void error(Throwable e, Iterable<Tuple> tuples) {
+
     if(!Iterables.isEmpty(tuples)) {
-      LOG.error("Failing {} tuples", Iterables.size(tuples), e);
+      LOG.error("Failing tuples; count={}, error={}", Iterables.size(tuples), ExceptionUtils.getRootCauseMessage(e));
     }
-    tuples.forEach(t -> error.addRawMessage(messageGetStrategy.get(t)));
+    tuples.forEach(t -> collector.ack(t));
```

Comment:
wow, we did we really not ack here before?  That's a bug. good catch.

Reply:
Wait, @cestella isn't it already acked in line 123?

Reply:
AH yes, it is, but this is a new error method.  It feels that both `error` method should be calling another `error`method to do the basics like `ack()`.